src/jailer/src/env.rs

```diff
@@ -43,6 +43,17 @@ const DEV_URANDOM_WITH_NUL: &str = "/dev/urandom";
 const DEV_URANDOM_MAJOR: u32 = 1;
 const DEV_URANDOM_MINOR: u32 = 9;
 
+// Userfault file descriptor device path. This is a misc character device
+// with a MISC_DYNAMIC_MINOR minor device:
+// https://elixir.bootlin.com/linux/v6.1.51/source/fs/userfaultfd.c#L2176.
+//
+// This means that its minor device number will be allocated at run time,
+// so we will have to find it at initialization time parsing /proc/misc.
+// What we do know is the major number for misc devices:
+// https://elixir.bootlin.com/linux/v6.1.51/source/Documentation/admin-guide/devices.txt
+const DEV_UFFD_PATH: &str = "/dev/userfaultfd";
+const DEV_UFFD_MAJOR: u32 = 10;
```

Comment:
Couldn't we just `stat("/dev/userfaultfd)` and copy the major and minor from there?

On second thought, since the others are hardcoded it's probably better to just keep them that way. Copying would be simpler but could potentially open the door to some attack.

Reply:
Yeah, just looking at the host's /dev/userfaultfd and copying whatever major/minor it has, instead of going through /proc/misc. But I think your approach is safer.